test/unit/StakingPool/requestAllocation.js

```diff
@@ -0,0 +1,488 @@
+const { expect } = require('chai');
+const { ethers } = require('hardhat');
+const {
+  getCurrentTrancheId,
+  calculateBasePrice,
+  calculateSurgePremiums,
+  calculatePriceBump,
+  divCeil,
+  roundUpToNearestAllocationUnit,
+} = require('./helpers');
+const { mineNextBlock, setNextBlockTime } = require('../../utils/evm');
+const { daysToSeconds } = require('../../../lib/helpers');
+const { AddressZero, MaxUint256 } = ethers.constants;
+const { parseEther } = ethers.utils;
+const { BigNumber } = ethers;
+
+describe('requestAllocation', function () {
```

Comment:
i know this ticket focuses on pricing values

requestAllocation is a fairly complex function that has a lot of effects 

We need tests for all those 

`updateStoredAllocations`

the // add new rewards section

// remove previous rewards

not sure if we already have a ticket for this

Reply:
Yea agreed. I couldn't find a ticket for it yet, so I made one here: https://github.com/NexusMutual/smart-contracts/issues/582
Feel free to add anything I missed.